src/HomieNode.hpp

```diff
@@ -33,23 +38,35 @@ class PropertyInterface {
 };
 
 class Property {
+  friend HomieNode;
   friend BootNormal;
 
  public:
-  explicit Property(const char* id, bool range = false, uint16_t lower = 0, uint16_t upper = 0) { _id = strdup(id); _range = range; _lower = lower; _upper = upper; _settable = false; }
+  explicit Property(const char* id) {
+    _id = strdup(id); _name = ""; _unit = ""; _datatype = ""; _format = ""; _retained = true; _settable = false; }
   void settable(const PropertyInputHandler& inputHandler) { _settable = true;  _inputHandler = inputHandler; }
+  void setName(const char* name) { _name = name; }
+  void setUnit(const char* unit) { _unit = unit; }
+  void setDatatype(const char* datatype) { _datatype = datatype; }
+  void setFormat(const char* format) { _format = format; }
+  void setRetained(const bool retained = true) { _retained = retained; }
+
 
  private:
-  const char* getProperty() const { return _id; }
+  const char* getId() const { return _id; }
+  const char* getName() const { return _name; }
+  const char* getUnit() const { return _unit; }
+  const char* getDatatype() const { return _datatype; }
+  const char* getFormat() const { return _format; }
+  bool isRetained() const { return _retained; }
   bool isSettable() const { return _settable; }
-  bool isRange() const { return _range; }
-  uint16_t getLower() const { return _lower; }
-  uint16_t getUpper() const { return _upper; }
   PropertyInputHandler getInputHandler() const { return _inputHandler; }
   const char* _id;
-  bool _range;
-  uint16_t _lower;
-  uint16_t _upper;
+  const char* _name;
+  const char* _unit;
+  const char* _datatype;
+  const char* _format;
```

Comment:
these should be stored as String, so it is automatically copied.

Of course most of the Strings that are set here a constants and thus in global memory. 
However, it is possible to receive dynamically created strings here and these could be on stack memory. For example, the allowed values for an ENUM could be generated dynamically.